### PR TITLE
fix: cleanup variable before re list and watch

### PR DESCRIPTION
**Pull request by zhixiongdu027**
### Description

Fixs #7489

### Checklist

- [] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by zhixiongdu027** [comment]
Hi, @tokers @crazyMonkey1995 
    
Please help review, Many tks.

File: apisix/discovery/kubernetes/informer_factory.lua [hunk 1]
```diff
@@ -263,6 +263,9 @@ local function list_watch(informer, apiserver)
     local reason, message
     local httpc = http.new()
 
+    informer.continue = ""
+    informer.version = ""
```
Comment (tokers) [inline apisix/discovery/kubernetes/informer_factory.lua:267]: Is it possible to add some cases to verify it?

Comment (zhixiongdu027) [reply apisix/discovery/kubernetes/informer_factory.lua:267]: I've thought about it too, but it's difficult.
Unless we are able to modify the value of "continue" at runtime in some way

Comment (crazyMonkey1995) [reply apisix/discovery/kubernetes/informer_factory.lua:267]: There is really no simpler way to reproduce. This unit test case is probably not easy to write.

**Comment by tokers** [comment]
I just re-ran the failed CI jobs.

**Comment by tzssangglass** [comment]
pls rebase the upstream master branch to your PR branch, the CI failed has been fixed in the master branch of upstream.

**Comment by zhixiongdu027** [comment]
@tzssangglass @tokers 
Please help re start ci .

**Comment by spacewander** [comment]
> @tzssangglass @tokers Please help re start ci .

Done

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**